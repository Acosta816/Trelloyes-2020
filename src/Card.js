import React from 'react';
import './Card.css';

//accepts props: title and content and id
function Card(props){
    console.log(props);

    return(
        <div className="Card">
            <button type="button" onClick={()=>{props.cbDeleteCard(props.id)}}>delete</button>
            <h3>
                {props.title}
            </h3>
            <p>
                {props.content}
            </p>
        </div>
    )
}

export default Card;