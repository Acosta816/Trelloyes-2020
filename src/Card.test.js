import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('TestSuite: Card Component', ()=> {

    it('SMOKE: renders component without crashing', ()=> {
        const div = document.createElement('div');

        ReactDOM.render(<Card />, div);

        ReactDOM.unmountComponentAtNode(div);
    });

    it('SNAPSHOT#1: renders the UI as expected compared to the saved shot', ()=> {
        const tree = renderer.create(<Card /> ).toJSON();

        expect(tree).toMatchSnapshot();
    })

})