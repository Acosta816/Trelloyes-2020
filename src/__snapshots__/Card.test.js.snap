// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite: Card Component SNAPSHOT#1: renders the UI as expected compared to the saved shot 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;
